src/Repository/CommitteeRepository.php

```diff
@@ -334,4 +337,53 @@ public function paginate(int $offset = 0, int $limit = SearchParametersFilter::D
 
         return new Paginator($query);
     }
+
+    public function retrieveMostActiveCommitteesInReferentManagedArea(Adherent $referent, int $limit = 5): array
+    {
+        return $this->retrieveTopCommitteesInReferentManagedArea($referent, $limit);
+    }
+
+    public function retrieveLeastActiveCommitteesInReferentManagedArea(Adherent $referent, int $limit = 5): array
+    {
+        return $this->retrieveTopCommitteesInReferentManagedArea($referent, $limit, false);
+    }
+
+    private function retrieveTopCommitteesInReferentManagedArea(Adherent $referent, int $limit = 5, bool $mostActive = true): array
+    {
+        if (!$referent->isReferent()) {
+            throw new \InvalidArgumentException('Adherent must be a referent.');
+        }
+
+        $result = $this->createQueryBuilder('committee')
+            ->select('committee.name, COUNT(event) AS events, SUM(event.participantsCount) as participants')
+            ->join(Event::class, 'event', Join::WITH, 'event.committee = committee.id')
+            ->join('committee.referentTags', 'tag')
+            ->where('tag.id IN (:tags)')
+            ->andWhere('committee.status = :status')
+            ->andWhere('event.beginAt >= :from')
+            ->andWhere('event.beginAt < :until')
+            ->setParameter('tags', $referent->getManagedArea()->getTags())
+            ->setParameter('status', Committee::APPROVED)
+            ->setParameter('from', (new Chronos('first day of this month'))->setTime(0, 0, 0))
+            ->setParameter('until', (new Chronos('first day of next month'))->setTime(0, 0, 0))
+            ->setMaxResults($limit)
+            ->orderBy('events', $mostActive ? 'DESC' : 'ASC')
```

Comment:
what do this order?

Reply:
To retrieve the most or the least active committees.